### PR TITLE
Support additional friendly names on OID

### LINKED ISSUE #36342: ECCurve friendly names are not cross platform
Consider the following reproduction on Linux:
```csharp
using System.Security.Cryptography;

[Theory]
[InlineData("nistP256")]   // fails
[InlineData("secp256r1")]  // fails
[InlineData("ECDSA_P256")] // passes
public static void Repro(string curveFriendlyName)
{
    var ecParameters = new ECParameters()
    {
        Curve = ECCurve.CreateFromFriendlyName(curveFriendlyName),
        Q = new ECPoint()
        {
            X = "65eda5a12577c2bae829437fe338701a10aaa375e1bb5b5de108de439c08551d".HexToByteArray(),
            Y = "1e52ed75701163f7f9e40ddf9f341b3dc9ba860af7e0ca7ca7e9eecd0084d19c".HexToByteArray(),
        },
    };

    ECDsa ecDsa = ECDsa.Create(ecParameters);
}
```
The first two identifiers will fail with the following exception:
```
System.PlatformNotSupportedException : The specified curve 'secp256r1' or its parameters are not valid for this platform.
Stack Trace:
    /home/eirik/devel/dotnet/runtime/src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EcDsa.ImportExport.cs(37,0): at Interop.Crypto.EcKeyCreateByKeyParameters(String oid, Byte[] qx, Int32 qxLength, Byte[] qy, Int32 qyLength, Byte[] d, Int32 dLength)
    /home/eirik/devel/dotnet/runtime/src/libraries/Common/src/System/Security/Cryptography/ECOpenSsl.ImportExport.cs(115,0): at System.Security.Cryptography.ECOpenSsl.ImportNamedCurveParameters(ECParameters parameters)
    /home/eirik/devel/dotnet/runtime/src/libraries/Common/src/System/Security/Cryptography/ECOpenSsl.ImportExport.cs(33,0): at System.Security.Cryptography.ECOpenSsl.ImportParameters(ECParameters parameters)
    /home/eirik/devel/dotnet/runtime/src/libraries/Common/src/System/Security/Cryptography/ECDsaOpenSsl.cs(317,0): at System.Security.Cryptography.ECDsaImplementation.ECDsaOpenSsl.ImportParameters(ECParameters parameters)
    /home/eirik/devel/dotnet/runtime/src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/ECDsaOpenSsl.cs(37,0): at System.Security.Cryptography.ECDsa.Create(ECParameters parameters)
    /home/eirik/devel/dotnet/runtime/src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CborWriterTests.cs(303,0): at System.Formats.Cbor.Tests.CborWriterTests.Repro(String curveFriendlyName)
```
Note that all three identifiers are recognized on Windows, so we should probably attempt to do the same in the OpenSSL implementation for the sake of consistency?

### PR DESCRIPTION
**Pull request by vcsjones**
Fixes #36342 

Adds the ability for OIDs to support mapping multiple friendly names on non-Windows platforms.

The previous design assumed there was a 1:1 for friendly name <-> OID. I'm not sure if this breaks assumptions elsewhere in the framework.

### CODE REVIEW

**Comment by vcsjones** [comment]
The first commit only changes the dictionary initializer to the cctor builder. The second commit adds additional friendly names.

File: src/libraries/System.Security.Cryptography.Encoding/src/Internal/Cryptography/OidLookup.cs [hunk 1]
```diff
@@ -237,29 +121,155 @@ internal static partial class OidLookup
                 { "1.3.14.7.2.3.1", "md2RSA" },
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography.Encoding/src/Internal/Cryptography/OidLookup.cs:121]: The new ones should just be added in this dictionary...

**Review by bartonjs — DISMISSED** [review]
There's already a second dictionary for "this is not the preferred return value for the OID", it should be used instead of a refactoring.

**Comment by vcsjones** [comment]
@bartonjs I'm not sure I understand. That dictionary doesn't get used for FriendlyName -> OID. It's used in `ToFriendlyName` which is used to resolve the friendly name from a given OID.

As far as I can tell, the purpose of that dictionary is to allow going from Oid -> FriendlyName, but not FriendlyName -> Oid. For example, on Linux this fails:

```
var oid = Oid.FromFriendlyName("RSA_KEYX", OidGroup.All);
Console.WriteLine(oid.Value);
```

But this will print RSA_KEYX:

```
var oid2 = Oid.FromOidValue("1.3.14.3.2.22", OidGroup.All);
Console.WriteLine(oid2.FriendlyName);
```

**Comment by bartonjs** [comment]
Ah.  OK, then I guess the refactor makes sense.  That one does seem to be for when there's more than one OID that points to the same friendly name (on Windows), not more than one friendly name that points to the same OID.  So it supported 1:m, not m:m.  Hmm.

**Review by bartonjs — APPROVED** [review]

**Pull request merged**